### PR TITLE
init push for backup wallet

### PR DESCRIPTION
**Pull request by Paladz**

### CODE REVIEW

File: blockchain/pseudohsm/image.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+func (h *HSM) Backup() ([]*KeyImage, error) {
+	images := []*KeyImage{}
+	xpubs := h.cache.keys()
+	for _, xpub := range xpubs {
+		xKey, err := ioutil.ReadFile(xpub.File)
+		if err != nil {
+			return nil, err
+		}
+
+		images = append(images, &KeyImage{XPub: xpub, XKey: xKey})
```
Comment (wliyongfeng) [inline blockchain/pseudohsm/image.go]: Maybe `xpub` is not necessary, while content of keystore file is enough for restore.

File: account/image.go [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+import (
+	"encoding/json"
+)
+
+type AccountSlice struct {
+	Account       *Account
+	ContractIndex uint64
+}
+
+type AccountImage struct {
```
Comment (wliyongfeng) [inline account/image.go]: `AccountImage` maybe not a good naming. Package name has 'account' info, so maybe just `Image` is enough.

File: asset/image.go [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+	assets := []*Asset{}
+	assetIter := reg.db.IteratorPrefix([]byte(assetPrefix))
+	defer assetIter.Release()
+
+	for assetIter.Next() {
+		asset := &Asset{}
+		if err := json.Unmarshal(assetIter.Value(), asset); err != nil {
+			return nil, err
+		}
+		assets = append(assets, asset)
```
Comment (wliyongfeng) [inline asset/image.go]: Maybe we don't need [RawDefinitionByte](RawDefinitionByte) in `Asset`. We don't have to export it while we can reduce the size.

File: asset/image.go [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		rawAsset, err := json.Marshal(asset)
+		if err != nil {
+			return err
+		}
+
+		storeBatch.Set(AliasKey(*asset.Alias), []byte(asset.AssetID.String()))
+		storeBatch.Set(Key(&asset.AssetID), rawAsset)
+	}
+	storeBatch.Write()
+	reg.saveNextAssetIndex(image.AssetIndex)
```
Comment (wliyongfeng) [inline asset/image.go]: Do we have to use backup this `AssetIndex`? We can find the max index of the backuped assets.

File: blockchain/pseudohsm/pseudohsm.go [hunk 5]
```diff
@@ -180,31 +180,3 @@ func (h *HSM) HasAlias(alias string) bool {
 func (h *HSM) HasKey(xprv chainkd.XPrv) bool {
 	return h.cache.hasKey(xprv.XPub())
 }
-
-//ImportXPrvKey import XPrv to chainkd
-func (h *HSM) ImportXPrvKey(auth string, alias string, xprv chainkd.XPrv) (*XPub, bool, error) {
```
Comment (wliyongfeng) [inline blockchain/pseudohsm/pseudohsm.go]: thx

File: blockchain/pseudohsm/image.go (hunk 1 above)
Comment (Paladz) [reply blockchain/pseudohsm/image.go]: keystore是加密的， 没密码读不到任何信息，我们导出时候不输密码

File: asset/image.go (hunk 3 above)
Comment (Paladz) [reply asset/image.go]: RawDefinitionByte这个东西像是原始数据，还是需要的。发资产的时候还是要靠RawDefinitionByte

File: asset/image.go (hunk 4 above)
Comment (Paladz) [reply asset/image.go]: 感觉这个还是需要的把？不然就可能会出现重资产的问题了？

File: account/image.go [hunk 6]
```diff
@@ -0,0 +1,86 @@
+// Package account stores and tracks accounts within a Chain Core.
```
Comment (Blockmeta) [inline account/image.go]: within a Bytom Core

**Pull request merged**